### ISSUE TITLE
the default grad_clip=None,so  when we should set optimizer_config=dict(grad_clip=dict(max_norm=1.0))?

### ISSUE THREAD

**Issue by pedroHuang123** [open-mmlab/mmflow#261]
Thanks for your error report and we appreciate it a lot.

**Checklist**

1. I have searched related issues but cannot get the expected help.
2. I have read the [FAQ documentation](https://mmdetection.readthedocs.io/en/latest/faq.html) but cannot get the expected help.
3. The bug has not been fixed in the latest version.

**Describe the bug**

**Reproduction**

1. What command or script did you run?

```none
A placeholder for the command.
```

2. Did you make any modifications on the code or config? Did you understand what you have modified?
3. What dataset did you use?

**Environment**

1. Please run `python mmdet/utils/collect_env.py` to collect necessary environment information and paste it here.
2. You may add addition that may be helpful for locating the problem, such as
   - How you installed PyTorch \[e.g., pip, conda, source\]
   - Other environment variables that may be related (such as `$PATH`, `$LD_LIBRARY_PATH`, `$PYTHONPATH`, etc.)

**Error traceback**
If applicable, paste the error trackback here.

```none
A placeholder for trackback.
```

**Bug fix**
If you have already identified the reason, you can provide the information here. If you are willing to create a PR to fix it, please also leave a comment here and that would be much appreciated!

**Comment by Zachary-66** [comment]
I don't think it's the implementation of `GaussianNoise` itself  leads to the explosion of `grad_norm`. In my opinion, the failure of fine tuning on our own datasets is very common.  Perhaps you can try some other super-parameters and lr schedulers?

**Comment by Zachary-66** [comment]
Yes, I have checked the config. At the same time, I checked the [implementation](https://github.com/open-mmlab/mmflow/blob/master/mmflow/datasets/pipelines/transforms.py#L1139) of `GaussianNoise`. There is nothing wrong with this implementation. Therefore, I concluded that it's not the implementation of `GaussianNoise` caused the explosion.
You can perform an ablation experiment (remove the `GaussianNoise`) to verify if there still exists explosion.

**Comment by Zachary-66** [comment]
There is no conflicts between `GaussianNoise` and `grad_clip`, so they can be set separately. Therefore, you don't have to set the `grad_clip` to `None`.
If you want to check the `grad_norm`, you can give the `max_norm` field in `grad_clip` a sufficiently large value, 100000, for example. Then the `grad_clip` won't effect the training while the `grad_norm` will be printed normally.

**Comment by Zachary-66** [comment]
As for the order of pipelines, maybe some specific transforms require certain input types. For example, `Normalize` are necessary before add `GaussianNoise`. As for the other transforms, you can study how they are implemented. There may be some slight differences caused by different order.

**Comment by Zachary-66** [comment]
I mean 200k iterations is time consuming, how did you make the experiment so fast. By saying more experiments, I mean you can try different lr scheduler, lr, image size, batch size, etc. This is a process of keep trying. I can't give you a specific answer, because the effect of deep learning needs to be verified by experiments, and I can't guarantee that my answer can improve your performance.
As for the `max_flow` field, we generally follow the setup in the original paper. Of course you can change its value to fit your dataset.
`Batch size` is an important parameter while fine tuning, so there might be some effects.

**Comment by Zachary-66** [comment]
A large `grad_norm` will effect the gradient back propagation, so I think it will effect the final results.
As for the use of `GaussianNoise`, the choices of hyper parameters, `sigma_range` and `clamp_range` also take effect. I think your experimental results are instructive. Because RAFT didn't use `GaussianNoise` but use `grad_clip`. While other models, such as FlowNet, they use `GaussianNoise` without `grad_clip`. Perhaps these authors met the same problems as yours.
Thanks for your issues and looking forward to your update!
